drivers/linux/amd64/cpu.asm

```diff
@@ -419,7 +419,11 @@ __swsmi__:
     mov rdi, r12 ; rdi_value
 
     ; this OUT instruction will write WORD value (smi_code_data) to ports 0xB2 and 0xB3 (SW SMI control and data ports)
-    out 0B2h, ax
+    push rax
+    shr ax, 8
+    out 0B3h, al
+    pop rax
+    out 0B2h, al

```

Comment:
This really looks worse that what it was before. Can we instead just add a comment that we're writing both ports at once?

Reply:
Ahh, that's actually some useful information to add to the commit log or even to the code?

Anyhow, I've some pending driver cleanups, also to this code (see https://github.com/minipli-oss/chipsec/commit/6f8f7f5777fbd2f7f80273c112f12040d4443ea4). But I can rebase these on top of your commit when it gets merged.

Reply:
Maybe we can use ROR instead which would avoid the PUSH, which seems to be wrongly annotated for the Windows version?:
```asm
    ror ax, 8
    out 0B3h, al ; 0xB3
    ror ax, 8
    out 0B2h, al ; 0xB2
```

Reply:
Using rotate is a much better idea to get rid of the push.

Reply:
changes made thanks for the help